[ENH] Add agents and claude md (#6176)
nilearn/nilearn: .github/pull_request_template.md

than work done entirely by humans,
it helps others to better trace eventual issues when reviewing the code.
-->
Use of AI tools:
- [ ] LLM tools were used to generate at least part of the content of this pull-request.

> - [ ] No LLM tools were used to generate at least part of the content of this pull-request.

Reviewer: ```suggestion
- [ ] No LLM tools were used to generate any part of the content of this pull-request.
```

Reviewer: This is not clear for me what exactly we want to understand with the second option.
